src/figuresClasses.ts

```diff
@@ -2,10 +1,93 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string;
+
+  color: string;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: string;
 
-export class Triangle implements Figure {}
+  color: string;
 
-export class Circle implements Figure {}
+  private radius: number;
 
-export class Rectangle implements Figure {}
+  constructor(color: string, radius: number) {
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: string;
+
+  color: string;
+
+  private width: number;
+
+  private height: number;
+
+  constructor(color: string, width: number, height: number) {
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
+    this.height = height;
+
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width and height must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(this.width * this.height * 100) / 100;
+  }
+}
 
-export function getInfo(figure): string {
-  return typeof figure;
+export function getInfo(figure: Figure): string {
+  return `A ${figure.color} ${figure.shape} - ${figure.getArea()}`;
```

Comment:
The `getInfo` function returns a string with the shape's color, type, and area. Consider adding units or more descriptive text for clarity, such as 'square units' after the area.